src/bsp/src/posix/src/TCPServer.cpp

```diff
@@ -96,7 +96,10 @@ void TCPServer::waitForClient() {
     // Always tries to reconnect to client
     while (m_serverFd > 0) {
         if (!m_connected) {
-            ::close(m_clientFd.value());
+            // Close socket
+            if (m_clientFd) {
+                ::close(m_clientFd.value());
```

Comment:
Only call close if the m_clientFd has a value. You should clear the optional afterwards